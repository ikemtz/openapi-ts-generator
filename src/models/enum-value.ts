export interface IEnumValue {
  key?: number;
  name: string;
  titleName: string;
  snakeCaseName: string;
}
