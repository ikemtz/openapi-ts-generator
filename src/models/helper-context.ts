import { IEntity } from './template-data';

export interface IHelperContext {
  data: {
    root: IEntity;
    key: number;
    index: number;
    first: boolean;
  };
  name: string;
}

export type PropertyType = 'valueProperties' | 'referenceProperties';
