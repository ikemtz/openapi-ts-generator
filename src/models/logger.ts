export interface ILogger {
  log: (data: string) => void;
  error: (data: string) => void;
  warn: (data: string) => void;
}

export class MockConsoleLogger implements ILogger {
  // tslint:disable-next-line: no-empty
  public log = (data: string) => {};
  // tslint:disable-next-line: no-empty
  public error = (data: string) => {};
  // tslint:disable-next-line: no-empty
  public warn = (data: string) => {};
}
