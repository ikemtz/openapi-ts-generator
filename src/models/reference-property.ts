export interface IReferenceProperty {
  name: string;
  snakeCaseName: string;
  referenceTypeName: string;
  typeScriptType: string;
  hasValidators: boolean;
  isSameAsParentTypescriptType: boolean;
  hasMultipleValidators: boolean;
  maxItems?: number;
  minItems?: number;
  maxLength?: number;
  minLength?: number;
  maximum?: number;
  minimum?: number;
  isArray: boolean;
  required: boolean;
  isEnum?: boolean;
  initialValue: string;
  initialTestValue: string;
}
