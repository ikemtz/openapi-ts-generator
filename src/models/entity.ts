import { IReferenceProperty } from './reference-property';
import { IValueProperty } from './value-property';
import { IImportType } from './template-data';

export interface IEntity {
  isEnum?: boolean;
  enumValues: (string | { key?: number; name: string; titleName: string })[];
  name: string;
  kebabCasedName: string;
  camelSingularName: string;
  description?: string;
  importTypes: IImportType[];
  valueProperties: IValueProperty[];
  referenceProperties: IReferenceProperty[];
}
