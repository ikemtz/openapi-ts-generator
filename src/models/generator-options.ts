/* eslint-disable @typescript-eslint/no-unused-vars */
import { resolve } from 'path';
import { ILogger } from './logger';
import { IEntity } from './entity';
import { IReferenceProperty } from './reference-property';
import { IValueProperty } from './value-property';

export interface IGeneratorOptions {
  logger?: ILogger;
  outputPath: string;
  openApiJsonUrl?: string;
  openApiJsonFileName?: string;
  genAngularFormGroups?: boolean;
  typeFilterCallBack?: (entity: IEntity, index: number, array: IEntity[]) => boolean;
  valuePropertyTypeFilterCallBack?: (valueProperty: IValueProperty, index: number, array: IValueProperty[]) => boolean;
  referencePropertyTypeFilterCallBack?: (referenceProperty: IReferenceProperty, index: number, array: IReferenceProperty[]) => boolean;
  pathUrlFormattingCallBack?: (url: string) => string;
  templates?: ITemplates | null;
}
export interface ITemplates {
  model?: string;
  testObjectFactory?: string;
  form?: string;
  formGroupFactory?: string;
  modelProperties?: string;
  barrel?: string;
  enum?: string;
  endpoints?: string;
}
export function defaultFilter(
  _value: IEntity | IValueProperty | IReferenceProperty,
  _index: number,
  _array: IEntity[] | IValueProperty[] | IReferenceProperty[],
): boolean {
  return true;
}

export function setGeneratorOptionDefaults(options: IGeneratorOptions): IGeneratorOptions {
  const templateFolder = resolve(`${__dirname}/..`, 'templates');
  options.typeFilterCallBack = options.typeFilterCallBack ?? defaultFilter;
  options.valuePropertyTypeFilterCallBack = options.valuePropertyTypeFilterCallBack ?? defaultFilter;
  options.referencePropertyTypeFilterCallBack = options.referencePropertyTypeFilterCallBack ?? defaultFilter;
  options.templates = {
    ...options.templates,
    model: options.templates?.model ?? `${templateFolder}/model.ts.hbs`,
    form: options.templates?.form ?? `${templateFolder}/form.ts.hbs`,
    formGroupFactory: options.templates?.formGroupFactory ?? `${templateFolder}/form-group-factory.ts.hbs`,
    testObjectFactory: options.templates?.testObjectFactory ?? `${templateFolder}/test-object-factory.ts.hbs`,
    barrel: options.templates?.barrel ?? `${templateFolder}/index.ts.hbs`,
    enum: options.templates?.barrel ?? `${templateFolder}/enum.ts.hbs`,
    modelProperties: options.templates?.barrel ?? `${templateFolder}/model-properties.ts.hbs`,
    endpoints: options.templates?.endpoints ?? `${templateFolder}/endpoints.ts.hbs`,
  };
  return options;
}
