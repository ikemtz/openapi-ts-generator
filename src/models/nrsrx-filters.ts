/* eslint-disable @typescript-eslint/no-unused-vars */
import { IEntity } from './entity';
import { IValueProperty } from './value-property';
const edmRegex = /^Edm[A-z]*Kind$/;

export function nrsrxTypeFilterCallBack(val: IEntity, _i: number, _arr: IEntity[]): boolean {
  return !val.name.endsWith('ODataEnvelope') && !edmRegex.test(val.name);
}
export function nrsrxValuePropertyTypeFilterCallBack(val: IValueProperty, _i: number, _arr: IValueProperty[]): boolean {
  return !val.name.startsWith('created') && !val.name.startsWith('updated');
}
