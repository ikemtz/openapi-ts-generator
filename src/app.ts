import { generateTsModels } from '.';
import { IGeneratorOptions } from './models/generator-options';
import { nrsrxTypeFilterCallBack, nrsrxValuePropertyTypeFilterCallBack } from './models/nrsrx-filters';
import { IEntity } from './models/template-data';

export const unitGenerationOptions = {
  openApiJsonUrl: 'https://d-unit-wal-00-cus-mstrcrp.azurewebsites.net/swagger/v1/swagger.json',
  outputPath: './output_unit/',
  typeFilterCallBack: nrsrxTypeFilterCallBack,
  valuePropertyTypeFilterCallBack: nrsrxValuePropertyTypeFilterCallBack,
};
export const messageGenerationOptions = {
  openApiJsonUrl: 'https://d-msng-wal-01-cus-mstrcrp.azurewebsites.net/swagger/v1/swagger.json',
  outputPath: './output_msng/',
  typeFilterCallBack: (val: IEntity, i: number, arr: IEntity[]) =>
    nrsrxTypeFilterCallBack(val, i, arr) && val.name !== 'GetMessageInfoResponse',
  valuePropertyTypeFilterCallBack: nrsrxValuePropertyTypeFilterCallBack,
};
export const accountGenerationOptions = {
  openApiJsonUrl: 'https://d-acct-wal-01-cus-mstrcrp.azurewebsites.net/swagger/v1/swagger.json',
  outputPath: './output_acct/',
  typeFilterCallBack: nrsrxTypeFilterCallBack,
  valuePropertyTypeFilterCallBack: nrsrxValuePropertyTypeFilterCallBack,
};

export const fileEmployeeGenerationOptions: IGeneratorOptions = {
  openApiJsonFileName: '../src/test-swagger-document.json',
  outputPath: './output_empl/',
  typeFilterCallBack: nrsrxTypeFilterCallBack,
  valuePropertyTypeFilterCallBack: nrsrxValuePropertyTypeFilterCallBack,
};

export function generateUnitFiles(options: IGeneratorOptions = unitGenerationOptions) {
  return generateTsModels(options);
}

export function generateMessageFiles(options: IGeneratorOptions = messageGenerationOptions) {
  return generateTsModels(messageGenerationOptions);
}

export function generateAccountFiles(options: IGeneratorOptions = accountGenerationOptions) {
  return generateTsModels(accountGenerationOptions);
}

export function generateEmployeeFiles(options: IGeneratorOptions = fileEmployeeGenerationOptions) {
  return generateTsModels(fileEmployeeGenerationOptions);
}

generateUnitFiles();
generateMessageFiles();
generateAccountFiles();
generateEmployeeFiles();
