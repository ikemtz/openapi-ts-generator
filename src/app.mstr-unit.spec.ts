import { mkdirSync, readdirSync } from 'fs';
import { generateTsModels, nrsrxTypeFilterCallBack, nrsrxValuePropertyTypeFilterCallBack } from '.';
import { ValidateFiles } from './app.spec';
import { IGeneratorOptions } from './models/generator-options';
import { MockConsoleLogger } from './models/logger';

const unitGenerationOptionsFactory = (): IGeneratorOptions => ({
  openApiJsonUrl: 'https://d-unit-wal-00-cus-mstrcrp.azurewebsites.net/swagger/v1/swagger.json',
  outputPath: './jest_output/unit/',
  typeFilterCallBack: nrsrxTypeFilterCallBack,
  valuePropertyTypeFilterCallBack: nrsrxValuePropertyTypeFilterCallBack,
});

describe('Url Based - Full Integration Tests', () => {
  describe('MasterCorp Unit Service', () => {
    it('should generate files', async (done) => {
      const options = unitGenerationOptionsFactory();
      try {
        mkdirSync(options.outputPath);
      } catch {
        // ignore
      }
      await generateTsModels(options);
      const files = readdirSync(options.outputPath).sort();
      ValidateFiles(options);
      expect(files).toMatchSnapshot();
      done();
    });

    it('should not generate files', async (done) => {
      const options = unitGenerationOptionsFactory();
      options.outputPath = './jest_output/unit_noFiles/';
      try {
        mkdirSync(options.outputPath);
      } catch {
        // ignore
      }

      await generateTsModels({
        ...options,
        logger: new MockConsoleLogger(),
        templates: {
          formGroupFactory: '',
          model: '',
          modelProperties: '',
          barrel: '',
          enum: '',
        },
      });
      ValidateFiles(options);
      done();
    });
  });
});
