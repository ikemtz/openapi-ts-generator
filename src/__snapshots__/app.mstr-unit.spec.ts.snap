// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Url Based - Full Integration Tests MasterCorp Unit Service should generate files 1`] = `
Array [
  "building.form-group-fac.ts",
  "building.model.ts",
  "building.properties.ts",
  "index.ts",
  "site-collection.form-group-fac.ts",
  "site-collection.model.ts",
  "site-collection.properties.ts",
  "site.form-group-fac.ts",
  "site.model.ts",
  "site.properties.ts",
  "tag.form-group-fac.ts",
  "tag.model.ts",
  "tag.properties.ts",
  "unit-tag.form-group-fac.ts",
  "unit-tag.model.ts",
  "unit-tag.properties.ts",
  "unit-type.form-group-fac.ts",
  "unit-type.model.ts",
  "unit-type.properties.ts",
  "unit.form-group-fac.ts",
  "unit.model.ts",
  "unit.properties.ts",
]
`;

exports[`Url Based - Full Integration Tests MasterCorp Unit Service should generate files: building.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { FormControl, FormGroup, Validators } from '@angular/forms';

export function BuildingFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    siteId: new FormControl(''),
    identifier: new FormControl(''),
    isEnabled: new FormControl(''),
    site: new FormControl(''),
    units: new FormControl(''),
  });
};
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Unit Service should generate files: building.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { ISite } from './site.model';
import { IUnit } from './unit.model';

export interface IBuilding {
  id?: string;
  siteId?: string;
  identifier?: string;
  isEnabled?: boolean;
  site?: ISite;
  units?: IUnit[];
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Unit Service should generate files: building.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model-properties.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum BuildingProperties {
  ID = 'id',
  SITE_ID = 'siteId',
  IDENTIFIER = 'identifier',
  IS_ENABLED = 'isEnabled',
  SITE = 'site',
  UNITS = 'units',
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Unit Service should generate files: index.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './building.form-group-fac';
export * from './building.model';
export * from './building.properties';
export * from './site-collection.form-group-fac';
export * from './site-collection.model';
export * from './site-collection.properties';
export * from './site.form-group-fac';
export * from './site.model';
export * from './site.properties';
export * from './tag.form-group-fac';
export * from './tag.model';
export * from './tag.properties';
export * from './unit-tag.form-group-fac';
export * from './unit-tag.model';
export * from './unit-tag.properties';
export * from './unit-type.form-group-fac';
export * from './unit-type.model';
export * from './unit-type.properties';
export * from './unit.form-group-fac';
export * from './unit.model';
export * from './unit.properties';
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Unit Service should generate files: site.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { FormControl, FormGroup, Validators } from '@angular/forms';

export function SiteFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    tenantId: new FormControl(''),
    accountId: new FormControl(''),
    accountName: new FormControl(''),
    siteCollectionId: new FormControl(''),
    number: new FormControl(''),
    name: new FormControl(''),
    addressLine1: new FormControl(''),
    addressLine2: new FormControl(''),
    city: new FormControl(''),
    state: new FormControl(''),
    zip: new FormControl(''),
    isEnabled: new FormControl(''),
    siteCollection: new FormControl(''),
    buildings: new FormControl(''),
    units: new FormControl(''),
  });
};
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Unit Service should generate files: site.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { ISiteCollection } from './site-collection.model';
import { IBuilding } from './building.model';
import { IUnit } from './unit.model';

export interface ISite {
  id?: string;
  tenantId?: string;
  accountId?: string;
  accountName?: string;
  siteCollectionId?: string;
  number?: string;
  name?: string;
  addressLine1?: string;
  addressLine2?: string;
  city?: string;
  state?: string;
  zip?: string;
  isEnabled?: boolean;
  siteCollection?: ISiteCollection;
  buildings?: IBuilding[];
  units?: IUnit[];
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Unit Service should generate files: site.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model-properties.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum SiteProperties {
  ID = 'id',
  TENANT_ID = 'tenantId',
  ACCOUNT_ID = 'accountId',
  ACCOUNT_NAME = 'accountName',
  SITE_COLLECTION_ID = 'siteCollectionId',
  NUMBER = 'number',
  NAME = 'name',
  ADDRESS_LINE_1 = 'addressLine1',
  ADDRESS_LINE_2 = 'addressLine2',
  CITY = 'city',
  STATE = 'state',
  ZIP = 'zip',
  IS_ENABLED = 'isEnabled',
  SITE_COLLECTION = 'siteCollection',
  BUILDINGS = 'buildings',
  UNITS = 'units',
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Unit Service should generate files: site-collection.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { FormControl, FormGroup, Validators } from '@angular/forms';

export function SiteCollectionFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    tenantId: new FormControl(''),
    number: new FormControl('', Validators.required),
    name: new FormControl('', Validators.required),
    code: new FormControl('', Validators.required),
    addressLine1: new FormControl(''),
    addressLine2: new FormControl(''),
    city: new FormControl(''),
    state: new FormControl('', Validators.compose([Validators.maxLength(2), Validators.minLength(2)])),
    zip: new FormControl('', Validators.compose([Validators.maxLength(10), Validators.minLength(5)])),
    isSchedulerEnabled: new FormControl('', Validators.required),
    siteCount: new FormControl(''),
    workCategoryTypes: new FormControl('', Validators.required),
    buildings: new FormControl(''),
    sites: new FormControl(''),
  });
};
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Unit Service should generate files: site-collection.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { IBuilding } from './building.model';
import { ISite } from './site.model';

export interface ISiteCollection {
  id?: string;
  tenantId?: string;
  number?: string;
  name?: string;
  code?: string;
  addressLine1?: string;
  addressLine2?: string;
  city?: string;
  state?: string;
  zip?: string;
  isSchedulerEnabled?: boolean;
  siteCount?: number;
  workCategoryTypes?: number;
  buildings?: IBuilding[];
  sites?: ISite[];
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Unit Service should generate files: site-collection.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model-properties.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum SiteCollectionProperties {
  ID = 'id',
  TENANT_ID = 'tenantId',
  NUMBER = 'number',
  NAME = 'name',
  CODE = 'code',
  ADDRESS_LINE_1 = 'addressLine1',
  ADDRESS_LINE_2 = 'addressLine2',
  CITY = 'city',
  STATE = 'state',
  ZIP = 'zip',
  IS_SCHEDULER_ENABLED = 'isSchedulerEnabled',
  SITE_COUNT = 'siteCount',
  WORK_CATEGORY_TYPES = 'workCategoryTypes',
  BUILDINGS = 'buildings',
  SITES = 'sites',
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Unit Service should generate files: tag.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { FormControl, FormGroup, Validators } from '@angular/forms';

export function TagFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    name: new FormControl(''),
    value: new FormControl(''),
    unitTags: new FormControl(''),
  });
};
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Unit Service should generate files: tag.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { IUnitTag } from './unit-tag.model';

export interface ITag {
  id?: string;
  name?: string;
  value?: string;
  unitTags?: IUnitTag[];
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Unit Service should generate files: tag.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model-properties.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum TagProperties {
  ID = 'id',
  NAME = 'name',
  VALUE = 'value',
  UNIT_TAGS = 'unitTags',
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Unit Service should generate files: unit.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { FormControl, FormGroup, Validators } from '@angular/forms';

export function UnitFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    number: new FormControl(''),
    unitTypeId: new FormControl(''),
    siteId: new FormControl(''),
    buildingId: new FormControl(''),
    isEnabled: new FormControl(''),
    building: new FormControl(''),
    site: new FormControl(''),
    unitType: new FormControl(''),
    unitTags: new FormControl(''),
  });
};
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Unit Service should generate files: unit.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { IBuilding } from './building.model';
import { ISite } from './site.model';
import { IUnitType } from './unit-type.model';
import { IUnitTag } from './unit-tag.model';

export interface IUnit {
  id?: string;
  number?: string;
  unitTypeId?: string;
  siteId?: string;
  buildingId?: string;
  isEnabled?: boolean;
  building?: IBuilding;
  site?: ISite;
  unitType?: IUnitType;
  unitTags?: IUnitTag[];
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Unit Service should generate files: unit.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model-properties.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum UnitProperties {
  ID = 'id',
  NUMBER = 'number',
  UNIT_TYPE_ID = 'unitTypeId',
  SITE_ID = 'siteId',
  BUILDING_ID = 'buildingId',
  IS_ENABLED = 'isEnabled',
  BUILDING = 'building',
  SITE = 'site',
  UNIT_TYPE = 'unitType',
  UNIT_TAGS = 'unitTags',
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Unit Service should generate files: unit-tag.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { FormControl, FormGroup, Validators } from '@angular/forms';

export function UnitTagFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    unitId: new FormControl(''),
    tagId: new FormControl(''),
    value: new FormControl(''),
    tag: new FormControl(''),
    unit: new FormControl(''),
  });
};
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Unit Service should generate files: unit-tag.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { ITag } from './tag.model';
import { IUnit } from './unit.model';

export interface IUnitTag {
  id?: string;
  unitId?: string;
  tagId?: string;
  value?: string;
  tag?: ITag;
  unit?: IUnit;
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Unit Service should generate files: unit-tag.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model-properties.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum UnitTagProperties {
  ID = 'id',
  UNIT_ID = 'unitId',
  TAG_ID = 'tagId',
  VALUE = 'value',
  TAG = 'tag',
  UNIT = 'unit',
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Unit Service should generate files: unit-type.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { FormControl, FormGroup, Validators } from '@angular/forms';

export function UnitTypeFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    code: new FormControl(''),
    description: new FormControl(''),
  });
};
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Unit Service should generate files: unit-type.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export interface IUnitType {
  id?: string;
  code?: string;
  description?: string;
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Unit Service should generate files: unit-type.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model-properties.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum UnitTypeProperties {
  ID = 'id',
  CODE = 'code',
  DESCRIPTION = 'description',
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Unit Service should not generate files 1`] = `Array []`;
